source/Calamari.Tests/AppServiceBehaviorFixture.cs

```diff
@@ -232,15 +264,29 @@ private static (string packagePath, string packageName, string packageVersion) P
                 ZipFile.CreateFromDirectory($"{tempPath.DirectoryPath}/AzureZipDeployPackage", packageinfo.packagePath);
                 return packageinfo;
             }
+            
+            private static (string packagePath, string packageName, string packageVersion) PrepareFunctionAppZipPackage()
+            {
+                (string packagePath, string packageName, string packageVersion) packageInfo;
+            
+                var testAssemblyLocation = new FileInfo(Assembly.GetExecutingAssembly().Location);
+                var sourceZip = Path.Combine(testAssemblyLocation.Directory.FullName, "functionapp.1.0.0.zip");
+                
+                packageInfo.packagePath = sourceZip;
+                packageInfo.packageVersion = "1.0.0";
+                packageInfo.packageName = "functionapp";
+            
+                return packageInfo;
+            }
 
-            private void AddVariables(CommandTestBuilderContext context)
+            private void AddVariables(CommandTestBuilderContext context, string siteName = null)
             {
                 context.Variables.Add(AccountVariables.ClientId, clientId);
                 context.Variables.Add(AccountVariables.Password, clientSecret);
                 context.Variables.Add(AccountVariables.TenantId, tenantId);
                 context.Variables.Add(AccountVariables.SubscriptionId, subscriptionId);
                 context.Variables.Add("Octopus.Action.Azure.ResourceGroupName", resourceGroupName);
-                context.Variables.Add("Octopus.Action.Azure.WebAppName", site.Name);
+                context.Variables.Add("Octopus.Action.Azure.WebAppName", siteName ?? site.Name);
```

Comment:
I can't find a place where we are making use of the optional parameter -  did I miss something?

Reply:
Ah. This will be from my initial attempt at writing a test with two Calamari processes in parallel. Good pickup 👍